python/cuspatial/cuspatial/core/geoseries.py

```diff
@@ -654,3 +688,140 @@ def _gather(
+
+        Parameters
+        ----------
+        other
+            a cuspatial.GeoSeries
+        align=True
+            to align the indices before computing .contains or not. If the
+            indices are not aligned, they will be compared based on their
+            implicit row order.
+
+        Examples
+        --------
+
+        Test if a polygon is inside another polygon:
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0.5, 0.5)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        dtype: bool
+
+
+        Test whether 3 points fall within either of two polygons
+        >>> gpdpoint = gpd.GeoSeries(
+            [Point(0, 0)],
+            [Point(-1, 0)],
+            [Point(-2, 0)],
+            [Point(0, 1)],
+            [Point(-1, 1)],
+            [Point(-2, 1)],
+            )
+        >>> gpdpolygon = gpd.GeoSeries(
+            [
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+                Polygon([[-2, -2], [-2, 2], [2, 2], [-2, -2]]),
+            ]
+        )
+        >>> point = cuspatial.from_geopandas(gpdpoint)
+        >>> polygon = cuspatial.from_geopandas(gpdpolygon)
+        >>> print(polygon.contains(point))
+        0    False
+        1    False
+        2    False
+        3     True
+        4     True
+        5    False
+        dtype: bool
+
+        Note
+        ----
+        input Series x and y will not be index aligned, but computed as
+        sequential arrays.
+
+        Note
+        ----
+        poly_ring_offsets must contain only the rings that make up the polygons
+        indexed by poly_offsets. If there are rings in poly_ring_offsets that
+        are not part of the polygons in poly_offsets, results are likely to be
+        incorrect and behavior is undefined.
+
+        Returns
+        -------
+        result : cudf.Series
+            A Series of boolean values indicating whether each point falls
+            within the corresponding polygon in the input.
+        """
+        if contains_only_polygons(self) is False:
+            raise TypeError("left series contains non-polygons.")
+
+        (lhs, rhs) = self.align(other) if align else (self, other)
+
+        # RHS conditioning:
+        mode = "POINTS"
+        # point in polygon
+        if contains_only_linestrings(rhs) is True:
+            # condition for linestrings
+            mode = "LINESTRINGS"
+            xy = rhs.lines
```

Comment:
I would rename `xy` to `geom` since it's the `GeoColumnAccessor`.

Reply:
ok